cdc_iterator.go

```diff
@@ -145,7 +148,9 @@ func (c *cdcIterator) Teardown(ctx context.Context) error {
 		//nolint:wrapcheck // no need to wrap canceled error
 		return ctx.Err()
 	case err := <-c.canalRunErrC:
-		if err != nil {
+		if errors.Is(err, replication.ErrSyncClosed) {
```

Comment:
We should log this error, so that logs can reveal why we stopped.

Reply:
Yeah, I agree. I don't return the error because in some tests that error was happening, even though the behaviour was as expected.